lms/djangoapps/teams/serializers.py

```diff
@@ -160,3 +162,6 @@ def __init__(self, *args, **kwargs):
         topics_to_team_count = {d['topic_id']: d['team_count'] for d in teams_per_topic}
         for topic in self.data['results']:
             topic['team_count'] = topics_to_team_count.get(topic['id'], 0)
+
+        if 'sort_order' in self.context and self.context['sort_order'] == 'team_count':
```

Comment:
Note to reviewers: sort_order was always being included in the context, but was ignored by this method. I chose to keep alphabetical sorting out of this method (as it was before), but am doing the team_count primary sort here.

Python documentation recommends doing two separate sorts in this sort of case instead of providing a more complicated comparator function that can handle the primary/secondary sorting as a single operation. 
https://docs.python.org/2/howto/sorting.html#sort-stability-and-complex-sorts